src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -96,6 +98,80 @@ int distance(BoundDecisionDagNode x)
             return result.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Enumerate the paths from the root node to the node of interest.
+        /// </summary>
+        /// <param name="nodes">The set of nodes in topological order.</param>
+        /// <param name="targetNode">The node of interest.</param>
+        /// <param name="nullPaths">Whether to permit following paths that test for null.</param>
+        /// <returns>Each path (excluding the node of interest) and whether it requires some when clause to evaluate to 'false'.</returns>
+        private static IEnumerable<(ImmutableArray<BoundDecisionDagNode> path, bool requiresFalseWhenClause)> GetPathsToNode(
```

Comment:
>GetPathsToNode [](http://example.com/codeflow?start=110&length=14)

Consider an alternative to using nested iterators, which shouldn't add to much complexity - pass a delegate that should be executed for each complete path we encounter during traversal. Returning `false` from the delegate should cancel the traversal.